offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/offergeneration/QuotationDetails.groovy

```diff
@@ -136,15 +176,28 @@ class QuotationDetails {
      */
     private void generateItemContent(OfferItem item, String elementId){
         itemNumber++
-        if (isOverflowingPage()) {
+
+        if (isTableOverflowByItem(item)) {
             generateHTMLTableOnNextPage(elementId)
             //repeat table header for next page
             htmlContent.getElementById(elementId).append(ItemPrintout.tableHeader())
             resetPageItemsCount()
+            consumedPageSpace += TableElementSpacing.TABLE_HEADER.getTableElementSpacing()
         }
         //add product to current table
         htmlContent.getElementById(elementId).append(ItemPrintout.itemInHTML(itemNumber, item))
-        pageItemsCount += determineItemSpace(item)
+        consumedPageSpace += determineItemSpace(item)
+    }
+
+    /**
+     * Checks if the addition of a new item would overflow the current table
+     * @param item The item to be appended to the current table
+     *
+     */
+    private boolean isTableOverflowByItem(OfferItem item) {
+
+        double itemSpace = determineItemSpace(item)
+        return isOverflowingPage(itemSpace)
```

Comment:
I think you dont need that method :)

Reply:
Ah right i can call the determineItem method in the classes. Smart 🧠